### ISSUE TITLE
preset particular functions to be turned ON by default

### ISSUE THREAD

**Issue by RichardPinka** [pyrevitlabs/pyRevit#2468]
## Is your feature request related to a problem? Please describe

tooggle buttons, i.e.  "Sync views"  , "Tab Coloring",  are always off after i turn them ON and restart revit

## Describe the solution you'd like

It would be perfect to have selected functionalities turned ON by default

**Comment by MohamedAsli** [comment]
Hi @RichardPinka,

It’s already the case for "Tab Coloring", since the variable is stored in the user settings file, which is read at Revit startup.
However, "Sync Views" is managed through pyRevit environment variables, which are session-specific, from what I understand.

This behavior can be changed, but I should ask:
1. Are you still interested in this feature?
2. If your answer is a strong "yes", I’ll need to dig deeper and check whether this change could lead to side effects

**Comment by jmcouffin** [comment]
Let me pitch in: 

- Defaulting the Tab coloring to ON would make sense as this is a feature most people use and activate in my experience
- Sync Views, this is not desirable as you rarely need to have the tool behavior always activated _ unless you are an electrical or mechanical engineer and all you do is placing vents and lights all day 😄 _ The Sync Views tool rely on events, which is a tad more processing power consuming.
- The distinction in the two behavior makes sense from my perspective.

**Comment by MohamedAsli** [comment]
Agree!

>Sync Views, this is not desirable as you rarely need to have the tool behavior always activated — unless you are an electrical or mechanical engineer and all you do is placing vents and lights all day 😄. The Sync Views tool relies on events, which is a tad more processing-power consuming.

Absolutely. That's what I was referring to when I said "side effects"

I'd also rather leave it as it is. But let’s wait for @RichardPinka to respond, I didn’t want to close the issue abruptly just to decrease the counter 😄

**Comment by RichardPinka** [comment]
dear @MohamedAsli  thank You for Your interest and motivation !  my answer is Yes, but actually, 
if i am the only one person on the planet who is keen to have this upgrade  (maybe some kind of votation would help You to find out what is preferable)  it is completely allright to keep it like it is...  i am actually just happy the function just works and that is allright. 

by the way, for example, valuable extra fucntion which i often think about is working on view property of attached model (see column C on picture below) (like an engineer who link - attach architectural model to background of my MEP model with dozen of floors) to be tabularly specify exact floor view from attached model to my view (i.e. either to do these settins manually on one place - on one table not each floor-view one by one like it is normal  workflow ; or to implement regular expressions able to write such rule :  choose the layout view from the list which will have in its name string "SLEP" AND string "objectAB" AND any number on the name will be same like the number of my layout view
automation ! :)

.... AND / OR  define wether this view will be "halftone" (i never used this)  or "underlay" (i often use this)  as columns A, B on pic below 

<img width="508" height="200" alt="Image" src="https://github.com/user-attachments/assets/c88075ca-c3b9-4506-be68-ab700d240af2" />


but I am fine and we all shall enjoy our lifes without time press and stress ;)

actually i am wondering to raise some money and create my own small tool which solve my own specific MEP design problems but these are just now in writing down the description function and drawing bpmn diagrams

**Comment by jmcouffin** [comment]
> but I am fine and we all shall enjoy our lifes without time press and stress ;)

😸

**Comment by RichardPinka** [comment]
but this is a need of any sub-profession engineer not architect.  (meaning the second suggestion) So I am not sure, wether it is interested to implement it . But imagine world, where each designer would be Architect, and 0 % of subprofession engineers !  would this world be working ?

**Comment by jmcouffin** [comment]
> but this is a need of any sub-profession engineer not architect. (meaning the second suggestion) So I am not sure, wether it is interested to implement it . But imagine world, where each designer would be Architect, and 0 % of subprofession engineers ! would this world be working ?

Ale v Cesko... 😄

**Comment by MohamedAsli** [comment]
> it is completely allright to keep it like it is...

If you're interested, I can share the script so you can apply it locally on your machine :)

> but this is a need of any sub-profession engineer not architect. (meaning the second suggestion) So I am not sure, wether it is interested to implement it . But imagine world, where each designer would be Architect, and 0 % of subprofession engineers ! would this world be working ?

I feel you 😅
But you know there's the pyRevitMEP extension of @CyrilWaechter, so I guess any good idea is worth trying. It can indeed be very useful

> Ale v Cesko... 😄

@jmcouffin fyi, used chatgpt to understand your message 🙃

**Comment by jmcouffin** [comment]
> [@jmcouffin](https://github.com/jmcouffin) fyi, used chatgpt to understand you message 🙃

yeah, that was directed to Richard as he lives in Czech Rep. like me.
What I was pointing at is the fact that engineers can stamp building permit drawing, they do not need to be architects. Which kind of sucks as they take a good share of the architects' jobs. Enough controversy!

**Comment by RichardPinka** [comment]
to react on @jmcouffin   (not sure how to cite text here)... 

> yeah, that was directed to Richard as he lives in Czech Rep. like me.
>What I was pointing at is the fact that engineers can stamp building permit drawing, they do not need to be architects. Which kind of sucks as they take a good share of the architects' jobs. Enough controversy!


I do not know other countries formal processes within documentation approval (by authorised designer´s  "round stamp") 
but in Czech republic  ..... 
of course, can "architect" profession  stamp and approve whole building documentation (including all subprofessions like sewerage pipiing, hot-cold water piping, ventilation, heating, cooling, electrical).... he is authorised to do this by law which is higher position on the designer hierarchy.  

And it is completely OK and common to do this in small family house projects...  

but in large buildings , like administrative buildings, hosptials, schools , flat buildings, ...  i do not know any story, any project or any person (architect) who would take on its own personal and professional risk to put stamp and author whole projects including all subprofession.  I would tell this would be his professional suicide relating whole branche of errors an mistakes could be done by anyone below him who would not have its own sub-profession authorisation.  O other designers (sub-professionists) who takes responsibility for their main part of building. 

so an authorised engineer has an anuthorisation on particular sub-rofession (e.g.  electrical systems) and he cannot authorise anything else (e.g. structural part of design, architectural, interior, cooling... )


i would not tell that subprofesionists would by this steal("take a good share of..") the jobs to architects. They just need to forward their professional focus to important details architects just dont know, and dont want to know and do ;)   
it is a must, a necessity. 

i am wondering this should be in each and every country the same. ;)

**Comment by jmcouffin** [comment]
> i am wondering this should be in each and every country the same. ;)

Agreed, and for the countries I know well, you sign only your own set, in your own area of expertise both in France and Canada.

**Comment by rusalov** [comment]
Hello, I´m an architect in a studio specialized in large residential buildings, and this feature of sync views is required for me to be on most of the time, due to the buildings being up to 30 stories high, so I need to jump back and forth diferent levels to check the architecture.
I would really appreciate to be able to set this sync views option on by default, because I have to manually turn it on mostly everyday.
Thank you.
